src/d3a/models/strategy/settlement_strategies/__init__.py

```diff
@@ -0,0 +1,142 @@
+from d3a_interface.constants_limits import ConstSettings
+from pendulum import duration
+
+from d3a.constants import SettlementTemplateStrategiesConstants
+from d3a.d3a_core.exceptions import MarketException
+from d3a.models.strategy.update_frequency import TemplateStrategyBidUpdater, \
+    TemplateStrategyOfferUpdater
+
+
+def settlement_market_strategy_factory(
+        initial_buying_rate: float = None, final_buying_rate: float = None,
+        initial_selling_rate: float = None, final_selling_rate: float = None
+):
+    if initial_selling_rate is None:
+        initial_selling_rate = SettlementTemplateStrategiesConstants.INITIAL_SELLING_RATE
+    if final_selling_rate is None:
+        final_selling_rate = SettlementTemplateStrategiesConstants.FINAL_SELLING_RATE
+    if initial_buying_rate is None:
+        initial_buying_rate = SettlementTemplateStrategiesConstants.INITIAL_BUYING_RATE
+    if final_buying_rate is None:
+        final_buying_rate = SettlementTemplateStrategiesConstants.FINAL_BUYING_RATE
+    if ConstSettings.GeneralSettings.ENABLE_SETTLEMENT_MARKETS:
+        return SettlementMarketStrategy(
+            initial_buying_rate, final_buying_rate,
+            initial_selling_rate, final_selling_rate)
+    else:
+        return SettlementMarketStrategyInterface(
+            initial_buying_rate, final_buying_rate,
+            initial_selling_rate, final_selling_rate
+        )
+
+
+class SettlementTemplateStrategyBidUpdater(TemplateStrategyBidUpdater):
+
+    @staticmethod
+    def get_all_markets(area):
+        return area.settlement_markets.values()
+
+
+class SettlementTemplateStrategyOfferUpdater(TemplateStrategyOfferUpdater):
+
+    @staticmethod
+    def get_all_markets(area):
+        return area.settlement_markets.values()
+
+
+class SettlementMarketStrategyInterface:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def event_market_cycle(self, strategy):
+        pass
+
+    def event_tick(self, strategy):
+        pass
+
+    def event_bid_traded(self, strategy, market_id, bid_trade):
```

Comment:
```suggestion
    def event_bid_traded(self, strategy: BaseStrategy, market_id: str, bid_trade: Trade):
```

As recommended in other [PR comments](https://github.com/gridsingularity/d3a-web/pull/1410#discussion_r696600945) to add type hint

Reply:
Will do so , thanks!